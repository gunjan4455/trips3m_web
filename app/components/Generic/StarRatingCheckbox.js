import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StarRatings from 'components/Review/StarRatings';

class StarRatingCheckbox extends Component {

  toggleCheckboxChange = () => {
    const { handleOnChange, id } = this.props;
    handleOnChange(id);
  };

  render() {
    const isChecked = this.props.isChecked || false;
    const { value, name, disabled } = this.props;
    return (
      <div className="col-xs-6 p0">
        <input
          type="checkbox"
          name={name}
          value={value}
          id={`filter_${name}_${value}`}
          className="checkbox-common"
          checked={isChecked}
          onChange={this.toggleCheckboxChange}
          disabled={disabled}
        />
        <label className="row row-" htmlFor={`filter_${name}_${value}`}>
          <div className="fleft">
            <StarRatings
              id={`label_${name}_${value}`}
              rating={Number(value)}
              total={5}
            />
          </div>
          {
            this.props.count !== undefined ?
            <span className='at_count hide'>[{this.props.count}]</span> :null
          }
        </label>
      </div>
    );
  }
}

StarRatingCheckbox.propTypes = {
  name: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  count: PropTypes.number
};

export default StarRatingCheckbox;
