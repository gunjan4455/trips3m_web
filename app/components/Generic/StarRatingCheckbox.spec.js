import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import StarRatingCheckbox from './StarRatingCheckbox';

configure({ adapter: new Adapter() });

const MockHandleOnChange = jest.fn();

const DEFAULT_PROPS = {
  id:2,
  isChecked:false,
  name:"hotel_ratings[]",
  title:"3 Star",
  value:"3",
  disabled:false,
  filterCountKey:"hotel_ratings_3",
  handleOnChange: MockHandleOnChange
};

const shallowWrapperStarRatingCheckbox = (props) => {
  return shallow(<StarRatingCheckbox {...props} />);
};

describe('StarRatingCheckbox module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperStarRatingCheckbox(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleOnChange call properly', () => {
    const wrapper = shallowWrapperStarRatingCheckbox(props);
    wrapper.instance().toggleCheckboxChange();
    expect(MockHandleOnChange).toHaveBeenCalledWith(2);
  });

  it('should not render packages count when count in props is not defined', () => {
    const wrapper = shallowWrapperStarRatingCheckbox(props);  
    expect(wrapper.find('.at_count')).toHaveLength(0);
  });

  it('should render packages count when count in props is defined', () => {
    props.count = 101;
    const wrapper = shallowWrapperStarRatingCheckbox(props);  
    expect(wrapper.find('.at_count')).toHaveLength(1);
  });

});
