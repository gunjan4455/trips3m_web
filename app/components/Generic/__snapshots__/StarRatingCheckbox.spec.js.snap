// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRatingCheckbox module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StarRatingCheckbox
    disabled={false}
    filterCountKey="hotel_ratings_3"
    handleOnChange={[MockFunction]}
    id={2}
    isChecked={false}
    name="hotel_ratings[]"
    title="3 Star"
    value="3"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          className="checkbox-common"
          disabled={false}
          id="filter_hotel_ratings[]_3"
          name="hotel_ratings[]"
          onChange={[Function]}
          type="checkbox"
          value="3"
        />,
        <label
          className="row row-"
          htmlFor="filter_hotel_ratings[]_3"
        >
          <div
            className="fleft"
          >
            <StarRatings
              id="label_hotel_ratings[]_3"
              rating={3}
              total={5}
            />
          </div>
        </label>,
      ],
      "className": "col-xs-6 p0",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "checkbox-common",
          "disabled": false,
          "id": "filter_hotel_ratings[]_3",
          "name": "hotel_ratings[]",
          "onChange": [Function],
          "type": "checkbox",
          "value": "3",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fleft"
            >
              <StarRatings
                id="label_hotel_ratings[]_3"
                rating={3}
                total={5}
              />
            </div>,
            null,
          ],
          "className": "row row-",
          "htmlFor": "filter_hotel_ratings[]_3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <StarRatings
                id="label_hotel_ratings[]_3"
                rating={3}
                total={5}
              />,
              "className": "fleft",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "label_hotel_ratings[]_3",
                "rating": 3,
                "total": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          null,
        ],
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            className="checkbox-common"
            disabled={false}
            id="filter_hotel_ratings[]_3"
            name="hotel_ratings[]"
            onChange={[Function]}
            type="checkbox"
            value="3"
          />,
          <label
            className="row row-"
            htmlFor="filter_hotel_ratings[]_3"
          >
            <div
              className="fleft"
            >
              <StarRatings
                id="label_hotel_ratings[]_3"
                rating={3}
                total={5}
              />
            </div>
          </label>,
        ],
        "className": "col-xs-6 p0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "checkbox-common",
            "disabled": false,
            "id": "filter_hotel_ratings[]_3",
            "name": "hotel_ratings[]",
            "onChange": [Function],
            "type": "checkbox",
            "value": "3",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fleft"
              >
                <StarRatings
                  id="label_hotel_ratings[]_3"
                  rating={3}
                  total={5}
                />
              </div>,
              null,
            ],
            "className": "row row-",
            "htmlFor": "filter_hotel_ratings[]_3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <StarRatings
                  id="label_hotel_ratings[]_3"
                  rating={3}
                  total={5}
                />,
                "className": "fleft",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": "label_hotel_ratings[]_3",
                  "rating": 3,
                  "total": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            null,
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
